fixtures/docstring-proc-macro/tests/bindings/test_docstring.py

```diff
@@ -0,0 +1,52 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Test namespace
+import uniffi_docstring_proc_macro
+assert uniffi_docstring_proc_macro.__doc__
+
+from uniffi_docstring_proc_macro import *
+
+# Test function
+assert test.__doc__ == "<docstring-function>"
+assert test_without_docstring.__doc__ is None
+
+# Test enums
+assert EnumTest.__doc__ == "<docstring-enum>"
+
+# Simple enum variants can't be tested, because `__doc__` is not supported for enums
+# assert EnumTest.ONE.__doc__ == "<docstring-enum-variant>"
+# assert EnumTest.TWO.__doc__ == "<docstring-enum-variant-2>"
+
+assert AssociatedEnumTest.__doc__ == "<docstring-associated-enum>"
+
+# `__doc__` is lost because of how enum templates are generated
+# https://github.com/mozilla/uniffi-rs/blob/eb97592f8c48a7f5cf02a94662b8b7861a6544f3/uniffi_bindgen/src/bindings/python/templates/EnumTemplate.py#L60
+# assert AssociatedEnumTest.TEST.__doc__ == "<docstring-associated-enum-variant>"
+# assert AssociatedEnumTest.TEST2.__doc__ == "<docstring-associated-enum-variant-2>"
+
+# Test errors
+assert ErrorTest.__doc__ == "<docstring-error>"
+assert ErrorTest.One.__doc__ == "<docstring-error-variant>"
+assert ErrorTest.Two.__doc__ == "<docstring-error-variant-2>"
+
+assert AssociatedErrorTest.__doc__ == "<docstring-associated-error>"
+assert AssociatedErrorTest.Test.__doc__ == "<docstring-associated-error-variant>"
+assert AssociatedErrorTest.Test2.__doc__ == "<docstring-associated-error-variant-2>"
+
+# Test objects
+assert ObjectTest.__doc__ == "<docstring-object>"
+assert ObjectTest.__init__.__doc__ == "<docstring-primary-constructor>"
+assert ObjectTest.new_alternate.__doc__ == "<docstring-alternate-constructor>"
+assert ObjectTest.test.__doc__ == "<docstring-method>"
+
+# Test records
+assert RecordTest.__doc__ == "<docstring-record>"
+
+# `__doc__` is not supported for class fields
```

Comment:
I can't test this on my project due to #1827, but to clarify the docs are still generated for the things with these comments, it's only that `__doc__` doesn't work so the test can't be done?

Reply:
I am not a Pythonista but as I understand it, it is a limitation of `__doc__`. After reading the comments, I did not try to improve the test on this.

Reply:
OK I got what you mean. So the answer is yes. You can look at the `EnumTemplate.py` changes I made in this PR, where I add a docstring to fields of associated enums. However it was not possible for errors.